### PR TITLE
Upgrade OkHttp to 2.4.

### LINKED ISSUE #490: Upgrade to OkHttp 2.4
Just released. The API changes are known to break us. Let's look at what it'll take to upgrade.

### PR DESCRIPTION
**Pull request by madongfly**
Fixes #490 

@ejona86

### CODE REVIEW

File: okhttp/src/main/java/com/squareup/okhttp/OkHttpTlsUpgrader.java [hunk 1]
```diff
@@ -56,10 +53,7 @@
   // TODO(madongfly): We should only support "h2" at a right time.
   private static final List<String> SUPPORTED_HTTP2_PROTOCOLS = Collections.unmodifiableList(
       Arrays.asList("h2", "h2-14", "h2-15", "h2-16"));
-
-  // A dummy address used to bypass null check.
-  private static final InetSocketAddress DUMMY_INET_SOCKET_ADDRESS =
-      InetSocketAddress.createUnresolved("fake", 73);
+  private static final List<Protocol> TLS_PROTOCOLS = Arrays.<Protocol>asList(Protocol.HTTP_2);
```
Comment (ejona86) [inline okhttp/src/main/java/com/squareup/okhttp/OkHttpTlsUpgrader.java]: This list is mutable, so it shouldn't be in all caps.

Better would be wrap it with collections.unmodifiablelist

Comment (madongfly) [inline okhttp/src/main/java/com/squareup/okhttp/OkHttpTlsUpgrader.java]: Done.

**Comment by madongfly** [comment]
@ejona86 PTAL, thanks!

**Comment by ejona86** [comment]
@madongfly LGTM

**Pull request merged**